src/dc_mimeparser.rs

```diff
@@ -804,30 +780,24 @@ impl<'a> MimeParser<'a> {
         msg_type: Viewtype,
         mime_type: libc::c_int,
         raw_mime: &str,
-        decoded_data: *const libc::c_char,
-        decoded_data_bytes: libc::size_t,
+        decoded_data: &[u8],
         desired_filename: &str,
     ) {
         /* create a free file name to use */
         let path_filename = dc_get_fine_path_filename(self.context, "$BLOBDIR", desired_filename);
-        let bytes = std::slice::from_raw_parts(decoded_data as *const u8, decoded_data_bytes);
 
         /* copy data to file */
-        if dc_write_file(self.context, &path_filename, bytes) {
+        if dc_write_file(self.context, &path_filename, decoded_data) {
             let mut part = Part::default();
             part.typ = msg_type;
             part.mimetype = mime_type;
-            part.bytes = decoded_data_bytes as libc::c_int;
+            part.bytes = decoded_data.len() as libc::c_int;
             part.param.set(Param::File, path_filename.to_string_lossy());
             part.param.set(Param::MimeType, raw_mime);
             if mime_type == 80 {
-                assert!(!decoded_data.is_null(), "invalid image data");
-                let data = std::slice::from_raw_parts(
-                    decoded_data as *const u8,
-                    decoded_data_bytes as usize,
-                );
+                assert!(!decoded_data.is_empty(), "invalid image data");
```

Comment:
isn't this crashing when there is a zero-length image file?

Reply:
yeah, I would suggest this method should be safeguarded against being called with empty data, writing an empty file seems like the wrong thing to do

Reply:
fixed